backend/lambdas/generators/json_report/json_report/results/secret.py

```diff
@@ -35,31 +50,50 @@ def get_secrets(scan, params):
         # If the scan was run with a diff set the diff_summary
         diff_summary = scan.diff_summary
 
+    filename_dict: FilenameDictType = {}
```

Comment:
The main rationale to moving towards this dict rather than continuing to directly edit the array is that we now need to de-duplicate based on a subset of properties.

For example, the following does not work, since `validity` should not be considered when de-duplicating:
```
result = [
  {
    "type": "...",
    "line": 1,
    "commit": "...",
    "validity": "unknown",
  }
]

# The same finding from another tool that does validity checks
item = {
  "type": "...",
  "line": 1,
  "commit": "...",
  "validity": "active",
}

if item not in result:
  # Same finding would be added multiple times
  result.append(item)
```